php/php.editor/src/org/netbeans/modules/php/editor/elements/PropertyHookElementImpl.java

```diff
@@ -0,0 +1,203 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.php.editor.elements;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.netbeans.modules.csl.api.OffsetRange;
+import org.netbeans.modules.csl.spi.ParserResult;
+import org.netbeans.modules.php.editor.CodeUtils;
+import org.netbeans.modules.php.editor.api.ElementQuery;
+import org.netbeans.modules.php.editor.api.PhpElementKind;
+import org.netbeans.modules.php.editor.api.elements.ParameterElement;
+import org.netbeans.modules.php.editor.api.elements.PropertyHookElement;
+import org.netbeans.modules.php.editor.model.impl.PropertyHookSignatureItem;
+import org.netbeans.modules.php.editor.model.nodes.PropertyHookDeclarationInfo;
+import org.netbeans.modules.php.editor.parser.astnodes.PropertyHookDeclaration;
+
+public class PropertyHookElementImpl extends PhpElementImpl implements PropertyHookElement {
+
+    private final boolean isReference;
+    private final boolean isAttributed;
+    private final boolean hasBody;
+    private final List<ParameterElement> parameters;
+    private final OffsetRange offsetRange;
+
+    private PropertyHookElementImpl(Builder builder) {
+        super(builder.name, builder.inScope, builder.filenameUrl, builder.offsetRange.getStart(), builder.elementQuery, builder.isDeprecated);
+        this.isReference = builder.isReference;
+        this.isAttributed = builder.isAttributed;
+        this.hasBody = builder.hasBody;
+        this.parameters = List.copyOf(builder.parameters);
+        this.offsetRange = builder.offsetRange;
+    }
+
+    @Override
+    public String getSignature() {
+        return PropertyHookSignatureItem.getSignatureFromElements(List.of(this));
+    }
+
+    @Override
+    public PhpElementKind getPhpElementKind() {
+        return PhpElementKind.PROPERTY_HOOK;
+    }
+
+    @Override
+    public boolean isReference() {
+        return isReference;
+    }
+
+    @Override
+    public boolean isAttributed() {
+        return isAttributed;
+    }
+
+    @Override
+    public boolean hasBody() {
+        return hasBody;
+    }
+
+    @Override
+    public List<? extends ParameterElement> getParameters() {
+        return parameters;
+    }
+
+    @Override
+    public OffsetRange getOffsetRange() {
+        return offsetRange;
+    }
+
+    @Override
+    public OffsetRange getOffsetRange(ParserResult result) {
+        return getOffsetRange();
+    }
+
+    /**
+     * Create property hook elements form PropertyHookSignatureItems.
+     *
+     * @param signatureItems
+     * @param field
+     * @param filenameUrl
+     * @param elementQuery
+     * @return property hook elements
+     */
+    public static List<PropertyHookElement> fromSignatureItems(final List<PropertyHookSignatureItem> signatureItems, final String field, final String filenameUrl, final ElementQuery elementQuery) {
+        List<PropertyHookElement> elements = new ArrayList<>();
```

Comment:
Nitpick:

```suggestion
        List<PropertyHookElement> elements = new ArrayList<>(signatureItems.size());
```

Reply:
Will fix it.

Reply:
Fixed: https://github.com/apache/netbeans/compare/33472a7a8c8a28e0896c06e55d656b1e9a9cd648..b7eb3b4cbfc2e915ecc84335650f39ea145074e6